src/main/kotlin/com/mparticle/kits/AppboyKit.kt

```diff
@@ -216,6 +254,42 @@ open class AppboyKit : KitIntegration(), AttributeListener, CommerceListener,
 
     override fun logEvent(event: CommerceEvent): List<ReportingMessage> {
         val messages: MutableList<ReportingMessage> = LinkedList()
+
+        //For CommerceEvent, Event Name is not required to generate hash. So, it will be always null.
+        event.products?.forEach {
+            it.customAttributes?.let {
+                changeUserArray(it, CommerceEventUtils.getEventType(event), null, true)
```

Comment:
i don't think we're suppose to map custom attributes of products like this - does the ui even support mapping them? I think you mean to just have `event.customAttributes` here